android/PhysicalWeb/app/src/main/java/org/physical_web/physicalweb/NearbyBeaconsFragment.java

```diff
@@ -213,28 +235,38 @@ public void onUrlMetadataIconReceived(PwoMetadata pwoMetadata) {
   }
 
   @SuppressWarnings("deprecation")
-  private void scanLeDevice(final boolean enable) {
-    if (mIsScanRunning != enable) {
-      mIsScanRunning = enable;
-      // If we should start scanning
-      if (enable) {
-        // Stops scanning after the predefined scan time has elapsed.
-        mHandler.postDelayed(mScanTimeout, SCAN_TIME_MILLIS);
-        // Clear any stored url data
-        mUrlToPwoMetadata.clear();
-        mNearbyDeviceAdapter.clear();
-        // Start the scan
-        mScanStartTime = new Date().getTime();
-        mBluetoothAdapter.startLeScan(mLeScanCallback);
-        // If we should stop scanning
-      } else {
-        // Cancel the scan timeout callback if still active or else it may fire later.
-        mHandler.removeCallbacks(mScanTimeout);
-        // Stop the scan
-        mBluetoothAdapter.stopLeScan(mLeScanCallback);
-        mSwipeRefreshWidget.setRefreshing(false);
-      }
-    }
+  private void stopScanning() {
+    // Cancel the scan timeout callback if still active or else it may fire later.
+    mHandler.removeCallbacks(mFirstScanTimeout);
+    mHandler.removeCallbacks(mSecondScanTimeout);
+    mHandler.removeCallbacks(mThirdScanTimeout);
+    // Stop the scanners
+    mMdnsUrlDiscoverer.stopScanning();
+    mSsdpUrlDiscoverer.stopScanning();
+    mBluetoothAdapter.stopLeScan(mLeScanCallback);
+    // Change the display appropriately
+    mSwipeRefreshWidget.setRefreshing(false);
+    mScanningAnimationDrawable.stop();
+  }
+
+  @SuppressWarnings("deprecation")
+  private void startScanning() {
```

Comment:
Perhaps we could add a short description of the scanning process, particularly with respect to the three timeouts?